pkg/go/graph/weighted_graph.go

```diff
@@ -0,0 +1,107 @@
+package graph
+
+import (
+	"fmt"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"gonum.org/v1/gonum/graph/encoding"
+	"gonum.org/v1/gonum/graph/encoding/dot"
+	"gonum.org/v1/gonum/graph/multi"
+)
+
+type WeightedAuthorizationModelGraph struct {
+	*multi.DirectedGraph
+	drawingDirection DrawingDirection
+}
+
+//nolint: cyclop
+func NewWeightedAuthorizationModelGraph(model *openfgav1.AuthorizationModel) (*WeightedAuthorizationModelGraph, error) {
+	g, err := NewAuthorizationModelGraph(model)
```

Comment:
Why not use the existing builder?

Reply:
The existing builder's job is to figure out the edges, their directions, etc. This builder's job is to assign weights. Since both are fairly complex jobs i'd like to keep them separate. Makes testing easier and less code per file